src/black/__init__.py

```diff
@@ -1315,7 +1315,9 @@ def assert_equivalent(src: str, dst: str, *, pass_num: int = 1) -> None:
         src_ast = parse_ast(src)
     except Exception as exc:
         raise AssertionError(
-            f"cannot use --safe with this file; failed to parse source file: {exc}"
+            f"cannot use --safe with this file; failed to parse source file AST: {exc}"
+            f"This could be caused by running black with an older python version "
+            f"that does not support new syntax used in your source file."
```

Comment:
This will all get mushed into one line, which probably won't look great.

Not sure if it's worth it, but maybe we can raise a custom exception class here, and in the top-level code that displays errors to the user format it nicely.

Reply:
I agree it isn't ideal.
That is more than I can take on at this time though(Covid/childcare problems).